### PR TITLE
refactor: move create database to procedure

### LINKED ISSUE #3591: move create database to procedure
### What type of enhancement is this?

Refactor, Tech debt reduction

### What does the enhancement do?

We introduced multiple-level locks in the procedure, and all DDL operations should acquire all related locks during the execution. After implementing #3516, we should refactor the database creation into a procedure. 

### Implementation challenges

_No response_

### PR DESCRIPTION
**Pull request by CookiePieWw**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)

close #3591

## What's changed and what's your intention?

move `create database` to procedure

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

**Comment by CookiePieWw** [comment]
The greptime-proto related part has not been finished yet, so I create a draft pr. I'll create another pr to greptime-proto once finished.

BTW, I found the `CreateDatabaseExpr` already defined in greptime-proto, which contains different params compared to the original impl of `create database` like the map `option` and missing `catalog`. I wonder how I would deal with it in the procedure.

**Comment by CookiePieWw** [comment]
> If you don't mind, you can replace the `greptime-proto` version with [your PR's](https://github.com/GreptimeTeam/greptime-proto/pull/141) commit hash.

Got it. The pr is ready for review.

File: src/common/meta/src/rpc/ddl.rs [hunk 1]
```diff
@@ -201,6 +223,7 @@ impl TryFrom<SubmitDdlTaskRequest> for PbDdlTaskRequest {
 
                 Task::AlterTableTasks(PbAlterTableTasks { tasks })
             }
+            DdlTask::CreateDatabase(_) => todo!(),
```
Comment (WenyXu) [inline src/common/meta/src/rpc/ddl.rs:226]: It seems you forgot to implement this

**Review by WenyXu — COMMENTED** [review]
Nice job👍

File: src/common/meta/src/rpc/ddl.rs (hunk 1 above)
Comment (CookiePieWw) [reply src/common/meta/src/rpc/ddl.rs]: Updated

**Comment by CookiePieWw** [comment]
I ran several tests since CI failed and I found the errors are mostly caused by case sensitivity and invalid schema names. But when I tried to reproduce and debug it on my mysql client, the results seem to be right. I'm a little confused. Maybe the client does something to make it right?

**Review by WenyXu — APPROVED** [review]

**Comment by CookiePieWw** [comment]
> All tests passed. BTW, would like to add a fuzz test for database creation (in another PR).

Sure I'd like to.

**Comment by WenyXu** [comment]
> > All tests passed. BTW, would like to add a fuzz test for database creation (in another PR).
> 
> Sure I'd like to.

Have fun😙

File: Cargo.toml [hunk 2]
```diff
@@ -104,7 +104,7 @@ etcd-client = "0.12"
 fst = "0.4.7"
 futures = "0.3"
 futures-util = "0.3"
-greptime-proto = { git = "https://github.com/GreptimeTeam/greptime-proto.git", rev = "06f6297ff3cab578a1589741b504342fbad70453" }
+greptime-proto = { git = "https://github.com/GreptimeTeam/greptime-proto.git", rev = "86e143918b70fe8c7b11df0f2cfcea76934f6217" }
```
Comment (waynexia) [inline Cargo.toml:107]: This commit is on your forked branch. Should use the one in main instead:

```suggestion
greptime-proto = { git = "https://github.com/GreptimeTeam/greptime-proto.git", rev = "1bd2398b686e5ac6c1eef6daf615867ce27f75c1" }
```

File: Cargo.lock [hunk 3]
```diff
@@ -3780,7 +3780,7 @@ checksum = "d2fabcfbdc87f4758337ca535fb41a6d701b65693ce38287d856d1674551ec9b"
 [[package]]
 name = "greptime-proto"
 version = "0.1.0"
-source = "git+https://github.com/GreptimeTeam/greptime-proto.git?rev=06f6297ff3cab578a1589741b504342fbad70453#06f6297ff3cab578a1589741b504342fbad70453"
+source = "git+https://github.com/GreptimeTeam/greptime-proto.git?rev=86e143918b70fe8c7b11df0f2cfcea76934f6217#86e143918b70fe8c7b11df0f2cfcea76934f6217"
```
Comment (waynexia) [inline Cargo.lock:3783]: Corresponding change to `Cargo.toml`

```suggestion
source = "git+https://github.com/GreptimeTeam/greptime-proto.git?rev=1bd2398b686e5ac6c1eef6daf615867ce27f75c1#1bd2398b686e5ac6c1eef6daf615867ce27f75c1"
```

**Comment by shuiyisong** [comment]
Hi, nice work!
I've noticed you used `database_name` in [proto definition](https://github.com/GreptimeTeam/greptime-proto/pull/141/files#diff-0ed821e94d43a185b3d614664c9cb002098e18c785b4566c8387e8682d9c639aR74), while we tend to use `schema_name`(corresponding to `catalog_name`, like we do in other expr definitions).
It would be great if you could modify the proto and this pr accordingly, so the code is in align with others. 😃

File: Cargo.toml (hunk 2 above)
Comment (shuiyisong) [reply Cargo.toml:107]: Should we use `branch` instead of `rev`?

Comment (CookiePieWw) [reply Cargo.toml]: Updated

**Comment by CookiePieWw** [comment]
> Hi, nice work! I've noticed that you used `database_name` in the [proto definition](https://github.com/GreptimeTeam/greptime-proto/pull/141/files#diff-0ed821e94d43a185b3d614664c9cb002098e18c785b4566c8387e8682d9c639aR74), whereas we typically use `schema_name` (corresponding to `catalog_name`, as we do in other expression definitions). It would be great if you could modify the proto and this PR accordingly, so the code aligns with the others. 😃

Since the pr of greptime-proto has been merged, it may be a little bit complicated. Maybe we can leave it with some future clean-up work?

File: Cargo.toml (hunk 2 above)
Comment (waynexia) [reply Cargo.toml]: Depending on the branch may lead to unexpected updates (since the lock file is usually ignored in review).

**Pull request merged**